active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,311 @@
+      },
+      // Extensible resource
+      {
++       "deploymentId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Resources/deployments/my-deployment",
++       "extensionAlias": "k8s",
++       "extension": "Kubernetes",
++       "symbolicName": "myService",
++       "resourceType": "core/Service",
++       "apiVersion": "v1",
++       "identifiers": {
++         "metadata": {
++           "namespace": "default",
++           "name": "myService",
++         },
++         "serverHostHash": "60fd32871cbe255d5793ce9e6ebf628beba25224cde7104e6a302f474f2f656e"
++       }
+      }
+    ]
+  }
+}
+```
+
+### API changes for `GET /Microsoft.Resources/deployments/operations/{operationId}` and `GET /Microsoft.Resources/deployments/operations`
+
+#### Updating `targetResource` to include extensible resource and extension information
+
+To align with modifications made to the `Microsoft.Resources/deployments` API, the response body now incorporates the addition of `symbolicName`, `resourceType`, `apiVersion`, `identifiers`, and `extension`.
+
+```diff
+{
+  ...
+  "properties": {
+    "provisioningOperation": "Create",
+    "provisioningState": "Succeeded",
+    "targetResource": {
+      "resourceType": "core/Service",
++     "apiVersion": "v1",
++     "symbolicName": "myService",
++     "identifiers": {
++       "metadata": {
++         "namespace": "default",
++         "name": "myService",
++       },
++       "serverHostHash": "60fd32871cbe255d5793ce9e6ebf628beba25224cde7104e6a302f474f2f656e"
++     },
++     "extension": {
++       "alias": "k8s",
++       "name": "Kubernetes",
++       "version": "1.27.8"
++     }
+    }
+  }
+}
+```
+
+### Client side changes
+
+### Azure CLI and Azure PowerShell
+
+Updating the Azure CLI and Azure PowerShell to effectively communicate deployment information for extensible resources is crucial for providing users with clear and actionable insights. A practical approach to this could include enhancing the output with informative log messages that clearly summarize essential details. For instance, implementing a log message format such as:
+
+```
+VERBOSE: 00:00:00 - Extensible resource with symbolic name 'myService' of type 'core/Service@v1' has provisioning status: 'Succeeded'. The resource is provisioned by 'Kubernetes@1.27.8'.
+```
+
+## Drawbacks
+
+### Inconsistency between Azure resources and extensible resources
+
+The new properties are added exclusively for extensible resources, leaving Azure resources unchanged, which avoids introducing breaking changes to the API, SDK, and clients. However, this approach leads to an inconsistency in the schema between Azure resources and extensible resources. Such a disparity could potentially cause confusion, as users might expect a uniform schema across all resource kinds. Clear documentation might be required to manage user expectations and explain the rationale behind the differing schemas.
```

Comment:
We might need to consider what the behavior is for ARM resources that have been declared as pseudo-extensible reosurces. See [here](https://msazure.visualstudio.com/One/_git/AzureUX-Deployments?path=/src/Templates.Tests/TemplateSymbolicNameTests.cs&version=GBmaster&line=486&lineEnd=523&lineStartColumn=1&lineEndColumn=16&lineStyle=plain&_a=contents) for an example.

Reply:
I'm thinking the decision to not modify the Azure resource schema should be reevaluated. To me it is strange that particular case versus non-extensible case would produce different results in the API. 

I think the decision not to change the Azure resource schemas was driven by 2 points:
1. Prevent regression in downstream systems for non-extension deployments by preserving schema.
2. Complications from varying deployment feature status (symbolic resource names for example).

I can highlight this topic in the upcoming discussion meeting.

Reply:
Decision was to investigate adding properties only under the feature flag and new API version using data that may already be available in the code context. Properties that may not be so simple to add may be omitted as necessary.

Reply:
Adding symbolic names to ARM resources for the operation API is deferred for now because sourcing the data is not so easy. Deployment resource reference does not have this data and would require significant refactors to supply it.